### ISSUE TITLE
[Feature Request] - SetPoint command to thermostat Stelpro  STZW402+

### ISSUE THREAD

**Issue by zeuspl** [dannybloe/dzVents#39]
Hi dannybloe, awesome work !!!

I'm new to Domoticz, and I was trying to push a SetPoint value to my Stelpro  STZW402+ z-wave thermostat.
I've try a lot of device methods with Set attribut, but without success.

From Domoticz GUI I can Modify the mode of the thermostat "Heat" or "Heat Econ" and I can change the SetPoint value of the thermostat.  From the Blockly events, I was able to change the SetPoint value with the block : "Set SetPoint Thermostat - SetPoint to __ Degrees".

Here the link of the manual : [Stelpro STZW402+](http://www.stelpro.com/sites/default/files/fr-CA/guide_utilisateur/INS_STZW402_1215_EN_0.pdf) 
There is really not much informations in this manual, but these 3 sentences might help : 

**Z-Wave Basic command class**
The Z-Wave Basic command class is supported by the thermostat in the following way:
To trigger the Eco mode on the thermostat: Send it a Basic SET (0x00) command (Energy Saving Mode)
To trigger the Comfort mode on the thermostat: Send it a Basic SET (0xFF) command (Comfort Mode)

I've created a small script comparing the exterior temperature and the temperature of the house, to decide when to start heating. It's almost complete i'm only missing the heating command ;o)

And if you have any suggestion to be able count minutes backward. ie:  I want to start heating the house 2 hours before we wake-up in the morning. (I've manage to do it with if & else loop, but it's not clean)

Thanks a lot, 
zeuspl

**Comment by zeuspl** [comment]
Hi dannybloe, 

To give you a bit more details on the devices, I joint some screenshot.

**Dashboard :** 
![thermostat](https://cloud.githubusercontent.com/assets/25071923/21892395/a8abe7f4-d8a4-11e6-8e41-1440118acc48.JPG)


**Thermostat mode :**
![thermostat - mode](https://cloud.githubusercontent.com/assets/25071923/21892407/c692b0ae-d8a4-11e6-8929-2b2a147997c3.JPG)


**Thermostat SetPoint :** 
![thermostat - setpoint](https://cloud.githubusercontent.com/assets/25071923/21892432/df318676-d8a4-11e6-8061-1021a76b1e98.JPG)


**Thermostat Temperature Sensor :** 
![thermostat - tempsensor](https://cloud.githubusercontent.com/assets/25071923/21892450/f107caae-d8a4-11e6-875b-cc3199e27b72.JPG)

If you need something else let me know.
zeuspl

**Comment by zeuspl** [comment]
One last thing, 

On the Open-zwave github project you will find the xml file of the thermostat that is installed in my Domoticz.

[stzw402.xml](https://github.com/OpenZWave/open-zwave/tree/master/config/stelpro)

Regards, 
zeuspl

**Comment by zeuspl** [comment]
Sorry for this request...

I did it with this simple line :  domoticz.devices['Thermostat'].update(0,22)

Thanks, 
zeuspl

**Comment by dannybloe** [comment]
Yes that should work. This maps to the standard domoticz commands. However it would be nice to be able to use the setpoint methods of dzVents. I will add this when I can find the time.

**Comment by zeuspl** [comment]
Ok Great, 
In this case this information might help you.

**Dashboard :**
![thermostat](https://cloud.githubusercontent.com/assets/25071923/21941004/c3f729c4-d994-11e6-934d-554f2bcad0b9.JPG)


**Thermostat SetPoint :**
![thermostat - setpoint](https://cloud.githubusercontent.com/assets/25071923/21941024/dea49482-d994-11e6-9ee3-3733ac958cbb.JPG)


**Thermostat - Mode :**
![thermostat - mode](https://cloud.githubusercontent.com/assets/25071923/21941036/ebae2152-d994-11e6-8c94-570bf90840f3.JPG)


**Thermostat - TempSensor :**
![thermostat - tempsensor](https://cloud.githubusercontent.com/assets/25071923/21941057/fa4c50c6-d994-11e6-8c59-92c0302d6f15.JPG)


**Open-zwave config file :** 
[stzw402.xml](https://github.com/OpenZWave/open-zwave/tree/master/config/stelpro)

Thanks, 
zeuspl

**Comment by dannybloe** [comment]
I have temperature and setpoint working. Could you check what command is sent from the GUI when you set the mode (what url is called? See network tab in e.g. chrome developer console)

**Comment by zeuspl** [comment]
Is that the information you need ?

Set Mode - Heat econo to Heat
![set mode - heat econo to heat](https://cloud.githubusercontent.com/assets/25071923/26661305/df08963a-464a-11e7-84b7-73e07eba0fce.JPG)


Set Mode - Heat to Heat econo
![set mode - heat to heat econo](https://cloud.githubusercontent.com/assets/25071923/26661317/eee031ee-464a-11e7-8962-e2e358cea96f.JPG)


Thanks, 
zeuspl

**Comment by dannybloe** [comment]
Ok, could you come up with a 'generic' url with placeholders for the variable parts like id, name, mode.
